### PR TITLE
Introduce ABAP Language Version base functionality

### PR DESCRIPTION
**Pull request by ThomasPloski**

### CODE REVIEW

File: src/repo/zcl_abapgit_repo.clas.abap [hunk 1]
```diff
@@ -910,4 +922,11 @@ CLASS zcl_abapgit_repo IMPLEMENTATION.
   METHOD zif_abapgit_repo~set_dot_abapgit.
     set( is_dot_abapgit = io_dot_abapgit->get_data( ) ).
   ENDMETHOD.
+
+
+  METHOD check_abap_language_version.
+    IF NEW zcl_abapgit_abap_language_vers( )->is_import_allowed( io_repo = me iv_package = ms_data-package ) = abap_false.
```
Comment (larshp) [inline src/repo/zcl_abapgit_repo.clas.abap:928]: this is not valid 702 syntax

Comment (ThomasPloski) [reply src/repo/zcl_abapgit_repo.clas.abap]: resolved

**Comment by mbtools** [comment]
The types and constants are already included in 
https://github.com/abapGit/abapGit/blob/main/src/objects/aff_types/zif_abapgit_aff_types_v1.intf.abap 
(which is a copy of https://github.com/SAP/abap-file-formats/blob/main/file-formats/zif_aff_types_v1.intf.abap).

This adds them in `zif_abapgit_definitions` and `zcl_abapgit_abap_language_vers`. Can we keep them in one place?

File: src/utils/zcl_abapgit_abap_language_vers.clas.abap [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+        CALL METHOD lo_abap_language_version->('IF_ABAP_LANGUAGE_VERSION~GET_DEFAULT_VERSION')
+          EXPORTING
+            iv_object_type     = iv_object_type
+            iv_package_name    = iv_package
+          RECEIVING
+            rv_default_version = rv_allowed_abap_langu_version.
+
+      CATCH cx_root.
+        rv_allowed_abap_langu_version =  c_allowed_abap_langu_version-standard.
```
Comment (schneidermic0) [inline src/utils/zcl_abapgit_abap_language_vers.clas.abap:108]: I guess, here we have to differentiate between source code object and non-source code objects, don't we?

**Review by schneidermic0 — COMMENTED** [review]
See my comments

**Comment by ThomasPloski** [comment]
> zif_abapgit_definitions



> The types and constants are already included in https://github.com/abapGit/abapGit/blob/main/src/objects/aff_types/zif_abapgit_aff_types_v1.intf.abap (which is a copy of https://github.com/SAP/abap-file-formats/blob/main/file-formats/zif_aff_types_v1.intf.abap).
> 
> This adds them in `zif_abapgit_definitions` and `zcl_abapgit_abap_language_vers`. Can we keep them in one place?

I have reused the already defined constants in zif_abapgit_aff_types_v1=>ty_abap_language_version.
Since the introduced ABAP Language Version in .abapgit.xml contains of semantically readable values like standard, keyUser, cloudDevelopment), I have decided to let the definition in the zif_abapgit_dot_abapgit interface.

File: src/utils/zcl_abapgit_abap_language_vers.clas.abap (hunk 2 above)
Comment (schneidermic0) [reply src/utils/zcl_abapgit_abap_language_vers.clas.abap]: @ThomasPloski and I discussed this. We try put the logic to the object handlers that need a "special" standard ABAP language version. If this won't work out, we add this logic later

**Review by schneidermic0 — APPROVED** [review]
Changes look good to me, now.

**Comment by larshp** [comment]
heh, so the changes from main combined with the changes in this PR gives a linter error, both changes independent no error 😄 dont think I've seen this before, but yea, possible

File: src/objects/zcl_abapgit_objects.clas.abap [hunk 3]
```diff
@@ -768,6 +771,10 @@ CLASS ZCL_ABAPGIT_OBJECTS IMPLEMENTATION.
             <ls_deser>-obj     = li_obj.
             <ls_deser>-xml     = lo_xml.
             <ls_deser>-package = lv_package.
+            <ls_deser>-item-abap_language_version = lo_abap_language_vers->get_abap_language_vers_by_objt(
```
Comment (mbtools) [inline src/objects/zcl_abapgit_objects.clas.abap:774]: Let's move this assigment up to right after `ls_item-devclass = lv_package.` (line 108). This way ALaV will be passed properly to the constructor via `create_object`.

File: src/utils/zcl_abapgit_abap_language_vers.clas.abap [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  PUBLIC
+  FINAL
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+
+    METHODS get_abap_language_vers_by_objt
+      IMPORTING
+        !iv_object_type                      TYPE trobjtype
+        !io_repo                             TYPE REF TO zif_abapgit_repo
```
Comment (mbtools) [inline src/utils/zcl_abapgit_abap_language_vers.clas.abap:11]: remove unused parameter `io_repo`

File: src/utils/zcl_abapgit_abap_language_vers.clas.abap [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    ENDIF.
+  ENDMETHOD.
+
+
+  METHOD is_import_allowed.
+
+    CASE get_abap_language_vers_by_repo( io_repo ).
+      WHEN zif_abapgit_dot_abapgit=>c_abap_language_version-undefined.
+        rv_allowed = abap_true.
+        RETURN.
```
Comment (mbtools) [inline src/utils/zcl_abapgit_abap_language_vers.clas.abap:115]: remove all 'RETURN`s

**Review by mbtools — CHANGES_REQUESTED** [review]
Some minor changes. The deserialize method became too long. We will refactor in separate PR later. Just set the check to 110 in your PR.

https://github.com/abapGit/abapGit/blob/9075658c3f0998ee859b6c319950179c9a8564c3/abaplint.json#L354

File: src/objects/zcl_abapgit_objects.clas.abap [hunk 6]
```diff
@@ -713,6 +716,8 @@ CLASS ZCL_ABAPGIT_OBJECTS IMPLEMENTATION.
           ENDIF.
 
           ls_item-devclass = lv_package.
+          ls_item-abap_language_version = lo_abap_language_vers->get_abap_language_vers_by_objt( iv_object_type = ls_item-obj_type
+                                                                                                 iv_package = lv_package ).
```
Comment (larshp) [inline src/objects/zcl_abapgit_objects.clas.abap:720]:
```suggestion
          ls_item-abap_language_version = lo_abap_language_vers->get_abap_language_vers_by_objt( 
            iv_object_type = ls_item-obj_type
            iv_package = lv_package ).
```

**Comment by mbtools** [comment]
I tested it on 702 with https://github.com/abapGit-tests/DDIC_integration (which contains core object types like `clas` `prog`, `fugr` as well). Works as usual. 

good to merge from my pov

**Comment by larshp** [comment]
@mbtools dismiss or approve your review?

**Review by mbtools — APPROVED** [review]

**Pull request merged**